### ISSUE TITLE
Can't retrieve strong-pm status from remote

### ISSUE THREAD

**Issue by cebor** [strongloop/strong-pm#279]
Since i updated node to v4.1.0 on remote and local machine the `status` cmd does not work anymore.

On local machine:
```bash
$ slc ctl -C http+ssh://user:password@stkn.org:8701 status
```

```
Service ID: 1
Service Name: loggrio-api
Environment variables:
  No environment variables defined
Instances:
    Version  Agent version  Cluster size  Driver metadata
Not started
```

On Remote:
```
$ sudo systemctl status strong-pm.service
```

```
● strong-pm.service - StrongLoop Process Manager
   Loaded: loaded (/etc/systemd/system/strong-pm.service; enabled; vendor preset: enabled)
   Active: active (running) since Mon 2015-09-21 10:52:02 CEST; 13min ago
 Main PID: 1515 (nodejs)
   CGroup: /system.slice/strong-pm.service
           ├─1515 /usr/bin/nodejs /usr/lib/node_modules/strong-pm/bin/sl-pm.js --listen 8701 --base /var/lib/strong-pm --base-port 3000 --driver direct
           ├─1623 /usr/bin/nodejs /usr/lib/node_modules/strong-pm/node_modules/strong-runner/node_modules/strong-supervisor/bin/sl-run.js --cluster=CPU --control=w...
           └─1628 /usr/bin/nodejs /usr/lib/node_modules/strong-pm/node_modules/strong-runner/node_modules/strong-supervisor/bin/sl-run.js ../e2a89e138a017ef54c1d7c...

Sep 21 10:57:42 stkn.org nodejs[1515]: 2015-09-21T08:57:42.767Z pid:1623 worker:0 INFO supervisor size set to 1
Sep 21 10:57:42 stkn.org nodejs[1515]: 2015-09-21T08:57:42.875Z pid:1623 worker:0 INFO supervisor started worker 1 (pid 1628)
Sep 21 10:57:42 stkn.org nodejs[1515]: 2015-09-21T08:57:42.877Z pid:1623 worker:0 INFO supervisor resized to 1
Sep 21 10:57:43 stkn.org nodejs[1515]: 2015-09-21T08:57:43.207Z pid:1628 worker:1 INFO strong-agent v2.0.1 profiling app 'loggrio-api' pid '1628'
Sep 21 10:57:43 stkn.org nodejs[1515]: 2015-09-21T08:57:43.213Z pid:1628 worker:1 INFO strong-agent[1628] started profiling agent
Sep 21 10:57:44 stkn.org nodejs[1515]: 2015-09-21T08:57:44.641Z pid:1628 worker:1 Swagger: skipping unknown type "ObjectID".
Sep 21 10:57:44 stkn.org nodejs[1515]: 2015-09-21T08:57:44.651Z pid:1628 worker:1 INFO strong-agent not profiling, agent metrics requires a valid license.
Sep 21 10:57:44 stkn.org nodejs[1515]: 2015-09-21T08:57:44.651Z pid:1628 worker:1 Please contact sales@strongloop.com for assistance.
Sep 21 10:57:44 stkn.org nodejs[1515]: 2015-09-21T08:57:44.661Z pid:1628 worker:1 Web server listening at: http://0.0.0.0:3001
Sep 21 10:57:44 stkn.org nodejs[1515]: 2015-09-21T08:57:44.662Z pid:1628 worker:1 Browse your REST API at http://0.0.0.0:3001/explorer
```
`restart`, `stop` and `start` cmds are working fine.

**Comment by cebor** [comment]
Also when rebooting the remote machine, the services don't come up again.

**Comment by robinskumar73** [comment]
I am having the same issue

**Comment by rmg** [comment]
@cebor when you upgraded node, did you re-install `strong-pm`? Did the version of strong-pm also get updated in the process?

@robinskumar73 same questions, but also: when you say "same issue" do you mean you are also using `systemd` and your app also does not come back up after rebooting? Or just that `slc ctl status` does not work?

**Comment by cebor** [comment]
@rmg yes its fresh install of strong-pm.

And with don't come up again, after reboot i do not mean the strong-pm service itself, but the services it handles.

**Comment by cebor** [comment]
Any updates? Is this a problem with node 4.1?

I'm sure that with node v0.12 all was fine.

**Comment by rmg** [comment]
@cebor sorry, I haven't been able to reproduce this.

For the small bit of log messages in your original posting, do the logs continue after that? Or does it get stuck? Could you post the contents of `/var/lib/strong-pm/strong-pm.json` somewhere? You can filter out the sensitive data and extra noise by running it through [jq](https://stedolan.github.io/jq/).

This should give the useful information but filter out anything sensitive:
```
cat /var/lib/strong-pm/strong-pm.json | jq '.ids + {ServiceProcess: .models.ServiceProcess} + {ServiceInstance: .models.ServiceInstance} + {Executor: .models.Executor} + {ServerService: .models.ServerService}'
```

**Comment by cebor** [comment]
```json
{
  "User": 1,
  "AccessToken": 1,
  "ServerService": {
    "1": "{\"name\":\"loggrio-api\",\"env\":{},\"_groups\":[{\"name\":\"default\",\"scale\":1,\"id\":1}],\"id\":1}"
  },
  "Group": 1,
  "Executor": {
    "1": "{\"address\":\"localhost\",\"hostname\":\"stkn.org\",\"metadata\":{},\"id\":1}"
  },
  "ServiceInstance": {
    "1": "{\"executorId\":1,\"serverServiceId\":1,\"groupId\":1,\"started\":false,\"setSize\":0,\"cpus\":\"CPU\",\"tracingEnabled\":false,\"id\":1}"
  },
  "ServiceProcess": {},
  "ServiceMetric": 1,
  "AgentTrace": 1,
  "ExpressUsageRecord": 1,
  "ProfileData": 1,
  "InstanceAction": 1
}
```

**Comment by rmg** [comment]
For anyone who finds this later, @cebor and I were able to sort this out over gitter.

The problem was caused by strong-pm telling the app to connect back to it using a public IP instead of localhost, but there were firewall policies in place that blocked such connections. This resulted in the app starting, and showing up in the logs, but being unable to connect to strong-pm in order to establish itself.

The fix was to use `--src-type LOCAL` instead of `-s localhost` for the `ACCEPT` rule so that a connection from the server's public IP to it's public IP was allowed.
